packages/slate/src/interfaces/path-ref.ts

```diff
@@ -21,18 +22,24 @@ export interface PathRefInterface {
 
 // eslint-disable-next-line no-redeclare
 export const PathRef: PathRefInterface = {
-  transform(ref: PathRef, op: Operation): void {
+  transform(ref: PathRef, op: Operation): (() => void) | void {
     const { current, affinity } = ref
 
     if (current == null) {
       return
     }
 
+    const prevPath = ref.current ? [...ref.current] : ref.current
```

Comment:
No need to duplicate the array. Although the `current` property is mutable, the path object itself never changes.
```suggestion
    const prevPath = ref.current
```